illumos does not need nor provide arc4random_addrandom
And evutil_secure_rng_add_bytes is a bad API anyway.

$OpenBSD: patch-include_event2_util_h,v 1.2 2015/01/06 21:28:05 sthen Exp $
--- a/include/event2/util.h.orig	Mon Jan  5 15:17:40 2015
+++ b/include/event2/util.h	Mon Jan  5 23:01:59 2015
@@ -690,22 +690,6 @@ int evutil_secure_rng_init(void);
  */
 int evutil_secure_rng_set_urandom_device_file(char *fname);
 
-/** Seed the random number generator with extra random bytes.
-
-    You should almost never need to call this function; it should be
-    sufficient to invoke evutil_secure_rng_init(), or let Libevent take
-    care of calling evutil_secure_rng_init() on its own.
-
-    If you call this function as a _replacement_ for the regular
-    entropy sources, then you need to be sure that your input
-    contains a fairly large amount of strong entropy.  Doing so is
-    notoriously hard: most people who try get it wrong.  Watch out!
-
-    @param dat a buffer full of a strong source of random numbers
-    @param datlen the number of bytes to read from datlen
- */
-void evutil_secure_rng_add_bytes(const char *dat, size_t datlen);
-
 #ifdef __cplusplus
 }
 #endif

$OpenBSD: patch-evutil_rand_c,v 1.3 2015/01/06 21:28:05 sthen Exp $
--- a/evutil_rand.c.orig	Mon Jan  5 15:17:40 2015
+++ b/evutil_rand.c	Mon Jan  5 23:01:59 2015
@@ -171,10 +171,3 @@ evutil_secure_rng_get_bytes(void *buf, size_t n)
 	ev_arc4random_buf(buf, n);
 }
 
-void
-evutil_secure_rng_add_bytes(const char *buf, size_t n)
-{
-	arc4random_addrandom((unsigned char*)buf,
-	    n>(size_t)INT_MAX ? INT_MAX : (int)n);
-}
-
